### PR TITLE
base64 functions

### LINKED ISSUE #3032: TO_BASE64() | Return the argument converted to a base-64 string

### LINKED ISSUE #3034: FROM_BASE64() | Decode base64 encoded string and return result

### PR DESCRIPTION
**Pull request by mshauneu**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Impl `to_base64` and `from_base64` functions.

## Changelog

- New Feature

## Related Issues

Fixes #3032
Fixes #3034

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/functions/src/scalars/strings/base_64.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
```
Comment (sundy-li) [inline common/functions/src/scalars/strings/base_64.rs:5]: I think we can use String2String template function to reduce the code size.

**Comment by BohuTANG** [comment]
@mergify update

**Review by sundy-li — APPROVED** [review]

File: website/databend/mkdocs.yml [hunk 2]
```diff
@@ -223,6 +223,8 @@ nav:
               - RAND: sqlstatement/numeric-functions/rand.md
           - String Functions:
               - SUBSTRING: sqlstatement/string-functions/substring.md
+              - TO_BASE64: sqlstatement/string-functions/to_base64.md
```
Comment (BohuTANG) [inline website/databend/mkdocs.yml:226]: Good job

**Pull request merged**